File: python/triangle/triangle.py
def equilateral(sides):
    a, b, c = sides[0], sides[1], sides[2]
    s = (a + b - c) * (a + c - b) * (b + c - a)
    if ((a + b >= c) and (a + c >= b) and (b + c >= a)) and s != 0:
        if (a == b and b == c):
            return True
        else:
            return False
    else:
        return False

def isosceles(sides):
    a, b, c = sides[0], sides[1], sides[2]
    s = (a + b - c) * (a + c - b) * (b + c - a)
    if ((a + b >= c) and (a + c >= b) and (b + c >= a)) and s != 0:
        if (a == b or b == c or a == c):
            return True
        else:
            return False
    else:
        return False


def scalene(sides):
    a, b, c = sides[0], sides[1], sides[2]
    s = (a + b - c) * (a + c - b) * (b + c - a)
    if ((a + b >= c) and (a + c >= b) and (b + c >= a)) and s != 0:
        if (a != b and b != c and a != c):
            return True
        else:
            return False
    else:
        return False

File: cpp/triangle/triangle.cpp
#include "triangle.h"

namespace triangle
{
    flavor kind(double a, double b, double c)
    {
        // 삼각형 성립 조건 검사
        if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a)
        {
            throw std::domain_error("Not a triangle");
        }

        // 삼각형 유형 판별
        if (a == b && b == c)
        {
            return flavor::equilateral;
        }
        else if (a == b || b == c || a == c)
        {
            return flavor::isosceles;
        }
        else
        {
            return flavor::scalene;
        }
    }
} // namespace triangle
